release-notes/major12/12.0.0/spring-boot3-migration.md

```diff
@@ -0,0 +1,70 @@
+# Migrating to Spring Boot 3
+
+The steps below will describe most relevant changes to Valtimo.
+However, please also check the following guides when running into issues:
+- [Spring Boot 3 Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)
+- [Preparing for Spring Security 6](https://docs.spring.io/spring-security/reference/5.8/migration/index.html)
+- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html)
+
+As a result of the Spring Boot 3 upgrade, some other frameworks / dependencies needed to be upgraded as well. 
+Each chapter below describes the changes for the relevant framework or dependency.  
+
+## Spring Boot 3 / Spring 6
+
+- `spring.factories` have been [replaced](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#auto-configuration-files) by `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`
+  - Listed configuration classes need to be annotated with `@AutoConfiguration`. More info can be found [here](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.7-Release-Notes#changes-to-auto-configuration)
+- `@Controller` or `@RestController` annotations are [now mandatory](https://github.com/spring-projects/spring-framework/issues/22154)
+  - When declaring the controller beans via autoconfiguration, please also annotate the controller with `com.ritense.valtimo.contract.annotation.@SkipComponentScan` to prevent duplicated beans.
+- `javax` packages [changed](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee) to `jakarta`
+- `@ConstructingBinding` on  no longer supported/needed for `@ConfigurationProperties` classes.
+- Java 8 is no longer supported. Please use Java 17.
```

Comment:
This seems like an important thing. Can you move this to the top of the list?

Reply:
done